WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadStarter.kt

```diff
@@ -155,12 +155,15 @@ class UploadStarter @Inject constructor(
                             false
                         } else {
                             trackAutoUploadAction(action, it.status)
```

Comment:
Out of context suggestion: I think both the tracking and logging should happen outside of the `filter`. I'd refactor the code with something like this:

```
postStore.getPostsWithLocalChanges(site)
        .asSequence()
        .map { post ->
            val action = uploadActionUseCase.getAutoUploadAction(post, site)
            Pair(post, action)
        }
        .filter { (_, action) ->
            action != DO_NOTHING
        }
        .toList()
        .forEach { (post, action) ->
            trackAutoUploadAction(action, post.status)
            AppLog.d(
                    AppLog.T.POSTS,
                    "UploadStarter for post title: ${post.title}, action: $action"
            )
            dispatcher.dispatch(
                    UploadActionBuilder.newIncrementNumberOfAutoUploadAttemptsAction(
                            post
                    )
            )
            uploadServiceFacade.uploadPost(
                    context = context,
                    post = post,
                    trackAnalytics = false
            )
        }

```

Reply:
Great suggestion ;) -> done in 216a882e5423ff89f3c5a22283952e32d5194383